packages/commonwealth/client/scripts/views/components/sidebar/index.tsx

```diff
@@ -13,10 +14,34 @@ export type SidebarMenuName =
   | 'exploreCommunities';
 
 export const Sidebar = ({ isInsideCommunity }) => {
-  const { menuName } = useSidebarStore();
+  const {
+    menuName,
+    menuVisible,
+    setRecentlyUpdatedVisibility,
+    recentlyUpdatedVisibility,
+  } = useSidebarStore();
+  const [shouldRender, setShouldRender] = useState(menuVisible);
```

Comment:
![image](https://github.com/hicommonwealth/commonwealth/assets/14819225/29fce20c-95f3-4b3a-a4e2-16ac71628ad1)

dont understand why we need `setShouldRender`?